### PR TITLE
Update "PythonProjectResource" to "PythonAppResource"

### LINKED ISSUE #5020: Make referencing Node/Npm projects consistent with python and dotnet
## Background and Motivation

When adding a Python or Dotnet project, the API is `builder.AddProject` / `builder.AddPythonProject` - Npm/Node should probably be consistent and not AddNodeApp etc. Also I'm sure there is a reason they are two separate APIs, but... feels like we should have builder.AddJSProject("node") or something and abstract that

## Proposed API

no

## Usage Examples

builder.AddNodeProject("pls");

### PR DESCRIPTION
**Pull request by maddymontaquila**
## Description

For .NET 9, we should obsolete `AddPythonProject` and replace it with `AddPythonApp` to align with `AddNodeApp` etc. the "project" term should be reserved for .NET projects, ie with a .csproj, to keep the differences between a "ProjectResource" and "ExecutibleResource" clear.

Notes - 
- I created a fully new class and methods to preserve the old ones as obsoleted. 
- I changed the summary of the api docs for anything obsoleted to say a generic thing we say in other .NET methods we obsolete.
- I also updated the tests - I didn't create new ones and leave the current ones, which I could do instead if that's better - and the playground.
- I left "project" as a term in api docs and params where it makes sense, ie the "python project's folder", because it seems like overkill to change all of that? Also at that point its really the python app entry points folder which is weird? But all of the methods and the class itself now use the terminology "App"

I fully expect there to be edits on this since it's the first time I'm touching an Aspire Integration sooooooooo go nuts!

Fixes #5020 (in the opposite way as requested, but per discussion with @aaronpowell and @DamianEdwards  this makes more sense)

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [x] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [x] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: TBD
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5759)

### CODE REVIEW

File: src/Aspire.Hosting.Python/PythonAppResourceBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    /// script and then use the <c>pip install -r requirements.txt</c> command to restore dependencies.
+    /// </para>
+    /// <para>
+    /// To receive traces, logs, and metrics from the python project in the dashboard, the project must be instrumented with OpenTelemetry.
+    /// You can instrument your project by adding the <c>opentelemetry-distro</c>, and <c>opentelemetry-exporter-otlp</c> to
+    /// your Python project.
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Add a python project to the application model. In this example the project is located in the <c>PythonProject</c> directory
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Python/PythonAppResourceBuilderExtensions.cs:47]: Perhaps reword some of this as well given the name/type change.

File: src/Aspire.Hosting.Python/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #nullable enable
-Aspire.Hosting.Python.PythonProjectResource
```
Comment (eerhardt) [inline src/Aspire.Hosting.Python/PublicAPI.Unshipped.txt:2]: @joperezr - do you know why all these APIs are still under "Unshipped" when we released an 8.2 version of these APIs?

The Shipped file says:

https://github.com/dotnet/aspire/blob/62f8cf792deee900b8afd04c8e9e278c7dc6fa8a/src/Aspire.Hosting.Python/PublicAPI.Shipped.txt#L1

Comment (maddymontaquila) [reply src/Aspire.Hosting.Python/PublicAPI.Unshipped.txt:2]: Oh yeah I was going to ask that lol

**Comment by maddymontaquila** [comment]
Ok so I didn't change the test syntax and now its failing but there was some weird spacing so maybe thats it? @joperezr pls halp

File: tests/Aspire.Hosting.Python.Tests/PythonPublicApiTests.cs [hunk 3]
```diff
@@ -173,8 +172,7 @@ public void AddPythonProjectWithVirtualEnvironmentPathShouldThrowWhenNameIsNull(
         const string virtualEnvironmentPath = ".venv";
         string[] scriptArgs = ["--traces"]; ;
 
-        var action = () => builder.AddPythonProject(
-            name,
+        var action = () App           name,
```
Comment (davidfowl) [inline tests/Aspire.Hosting.Python.Tests/PythonPublicApiTests.cs:175]: Something got deleted

**Comment by maddymontaquila** [comment]
GREAT NEWS these test failures aren't mine!!!

**Comment by davidfowl** [comment]
@radical the tests have gone bad again. Cosmos, Elastic search and timeouts all over.

**Comment by radical** [comment]
> @radical the tests have gone bad again. Cosmos, Elastic search and timeouts all over.

re:Elasticsearch, the tests need to be looked at properly. I see bunch of failures like:
```
      Watch task over Kubernetes Executable resources terminated unexpectedly. Check to ensure dcpd process is running.
      System.Net.Http.HttpRequestException: Error while copying content to a stream.
       ---> System.Net.Http.HttpProtocolException: The HTTP/2 server sent invalid data on the connection. HTTP/2 error code 'PROTOCOL_ERROR' (0x1). (HttpProtocolError)
         at System.Net.Http.Http2Connection.ThrowRequestAborted(Exception innerException)
         at System.Net.Http.Http2Connection.Http2Stream.TryReadFromBuffer(Span`1 buffer, Boolean partOfSyncRead)
         at System.Net.Http.Http2Connection.Http2Stream.ReadDataAsync(Memory`1 buffer, HttpResponseMessage responseMessage, CancellationToken cancellationToken)
         at k8s.LineSeparatedHttpContent.CancelableStream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
         at System.IO.StreamReader.ReadBufferAsync(CancellationToken cancellationToken)
         at System.IO.StreamReader.ReadLineAsyncInternal(CancellationToken cancellationToken)
         at k8s.Watcher`1.<>c.<CreateWatchEventEnumerator>b__21_1[TR](Task`1 t)
         at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location ---
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      --- End of stack trace from previous location ---
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+MoveNext()
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 216
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 216
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_1`1.<<WatchResourceChanges>b__10>d.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 331
      --- End of stack trace from previous location ---
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_1`1.<<WatchResourceChanges>b__10>d.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 331
      --- End of stack trace from previous location ---
         at Polly.ResiliencePipeline.<>c.<<ExecuteAsync>b__3_0>d.MoveNext()
      --- End of stack trace from previous location ---
         at Polly.Outcome`1.GetResultOrRethrow()
         at Polly.ResiliencePipeline.ExecuteAsync(Func`2 callback, CancellationToken cancellationToken)
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_0.<<WatchResourceChanges>g__WatchKubernetesResourceAsync|3>d`1.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 329
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpContent.LoadIntoBufferAsyncCore(Task serializeToStreamTask, MemoryStream tempBuffer)
         at System.Net.Http.HttpContent.WaitAndReturnAsync[TState,TResult](Task waitTask, TState state, Func`2 returnFunc)
         at k8s.Kubernetes.CreateResultAsync[T](HttpRequestMessage httpRequest, HttpResponseMessage httpResponse, Nullable`1 watch, CancellationToken cancellationToken)
         at k8s.AbstractKubernetes.ICustomObjectsOperations_ListClusterCustomObjectWithHttpMessagesAsync[T](String group, String version, String plural, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)
         at k8s.AbstractKubernetes.k8s.ICustomObjectsOperations.ListClusterCustomObjectWithHttpMessagesAsync(String group, String version, String plural, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)
         at k8s.WatcherExt.<>c__DisplayClass1_0`2.<<MakeStreamReaderCreator>b__0>d.MoveNext()
      --- End of stack trace from previous location ---
         at k8s.Watcher`1.<>c.<CreateWatchEventEnumerator>b__21_1[TR](Task`1 t)
         at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location ---
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      --- End of stack trace from previous location ---
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+MoveNext()
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 216
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 216
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_1`1.<<WatchResourceChanges>b__10>d.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 331
      --- End of stack trace from previous location ---
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_1`1.<<WatchResourceChanges>b__10>d.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 331
      --- End of stack trace from previous location ---
         at Polly.ResiliencePipeline.<>c.<<ExecuteAsync>b__3_0>d.MoveNext()
      --- End of stack trace from previous location ---
         at Polly.Outcome`1.GetResultOrRethrow()
         at Polly.ResiliencePipeline.ExecuteAsync(Func`2 callback, CancellationToken cancellationToken)
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass40_0.<<WatchResourceChanges>g__WatchKubernetesResourceAsync|3>d`1.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 329
```
Maybe this should have failed the test, but it seems to keep going.

re:cosmos, that's always an issue :/ cc @eerhardt 

One of the reasons for timeouts is that sometimes you have lot of docker images being downloaded and increases the time taken per test! We need to figure out a solution for this one.

**Comment by radical** [comment]
re:elasticsearch, towards the end you get:
```
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0] Stopped watching Container resources.
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0] Stopped watching Service resources.
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0] Stopped watching Endpoint resources.
crit: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Watch task over Kubernetes Executable resources terminated unexpectedly. Check to ensure dcpd process is running.
      System.Net.Http.HttpRequestException: Error while copying content to a stream.
       ---> System.Net.Http.HttpProtocolException: The HTTP/2 server sent invalid data on the connection. HTTP/2 error code 'PROTOCOL_ERROR' (0x1). (HttpProtocolError)
...
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0] Stopped watching Executable resources
...
```

.. but we keep running with rns. We should fail operations in that.

**Comment by davidfowl** [comment]
That looks like dcp crashed or stopped running or something.

**Comment by davidfowl** [comment]
@radical that error is the same as the one from [this issue](https://github.com/dotnet/aspire/issues/5559).

cc @karolz-ms

File: src/Aspire.Hosting.Python/PublicAPI.Shipped.txt [hunk 4]
```diff
@@ -1 +1,12 @@
 #nullable enable
+Aspire.Hosting.Python.PythonAppResource
```
Comment (DamianEdwards) [inline src/Aspire.Hosting.Python/PublicAPI.Shipped.txt:2]: This shouldn't be in the shipped file should it as you're adding it in this PR?

**Comment by karolz-ms** [comment]
I will work with @radical  to see if we can narrow down these `HTTP/2 error code PROTOCOL_ERROR` errors

**Review by mitchdenny — APPROVED** [review]

**Pull request merged**